src/custom-element.js

```diff
@@ -343,1087 +343,1100 @@ function isInternalOrServiceNode(node) {
 
-    /** @private {?Element} */
-    this.loadingElement_ = null;
+      /** @private {!./size-list.SizeList|null|undefined} */
+      this.heightsList_ = undefined;
 
-    /** @private {?Element|undefined} */
-    this.overflowElement_ = undefined;
+      /**
+       * This element can be assigned by the {@link applyLayout_} to a child
+       * element that will be used to size this element.
+       * @private {?Element}
+       */
+      this.sizerElement_ = null;
 
-    // `opt_implementationClass` is only used for tests.
-    let Ctor = knownElements[this.name];
-    if (getMode().test && this.implementationClassForTesting) {
-      Ctor = this.implementationClassForTesting;
-    }
+      /** @private {boolean|undefined} */
+      this.loadingDisabled_ = undefined;
 
-    /** @private {!./base-element.BaseElement} */
-    this.implementation_ = new Ctor(this);
+      /** @private {boolean|undefined} */
+      this.loadingState_ = undefined;
 
-    /**
-     * An element always starts in a unupgraded state until it's added to DOM
-     * for the first time in which case it can be upgraded immediately or wait
-     * for script download or `upgradeCallback`.
-     * @private {!UpgradeState}
-     */
-    this.upgradeState_ = UpgradeState.NOT_UPGRADED;
+      /** @private {?Element} */
+      this.loadingContainer_ = null;
+
+      /** @private {?Element} */
+      this.loadingElement_ = null;
+
+      /** @private {?Element|undefined} */
+      this.overflowElement_ = undefined;
+
+      // `opt_implementationClass` is only used for tests.
+      let Ctor = knownElements[this.name];
+      if (getMode().test && this.implementationClassForTesting) {
+        Ctor = this.implementationClassForTesting;
+      }
+      /** @private {./base-element.BaseElement} */
+      this.implementation_ = new Ctor(this);
+
+      /**
+       * An element always starts in a unupgraded state until it's added to DOM
+       * for the first time in which case it can be upgraded immediately or wait
+       * for script download or `upgradeCallback`.
+       * @private {!UpgradeState}
+       */
+      this.upgradeState_ = UpgradeState.NOT_UPGRADED;
+
+      /**
+       * Action queue is initially created and kept around until the element
+       * is ready to send actions directly to the implementation.
+       * @private {?Array<!./service/action-impl.ActionInvocation>}
+       */
+      this.actionQueue_ = [];
+
+      /**
+       * Whether the element is in the template.
+       * @private {boolean|undefined}
+       */
+      this.isInTemplate_ = undefined;
+    }
 
     /**
-     * Action queue is initially created and kept around until the element
-     * is ready to send actions directly to the implementation.
-     * @private {?Array<!./service/action-impl.ActionInvocation>}
+     * The Custom Elements V0 equivalent to the constructor.
+     * @suppress {checkTypes}
      */
-    this.actionQueue_ = [];
+    createdCallback() {
+      this.init();
```

Comment:
Unfortunately that fails at the `self = super(self);` line in the constructor with the error:

> custom-element.js:374 Uncaught TypeError: Failed to construct 'HTMLElement': Please use the 'new' operator, this DOM object constructor cannot be called as a function.

This error is avoided during normal operation since the polyfill calls the constructor using `Reflect`. Also, using an `init()` method allows using `this` internally rather than the local `self` var, which is more natural IMO.